wai-extra/Network/Wai/Middleware/Gzip.hs

```diff
@@ -92,13 +94,23 @@ gzip set app env sendResponse = app env $ \res ->
         ResponseFile{} | gzipFiles set == GzipIgnore -> sendResponse res
         _ -> if "gzip" `elem` enc && not isMSIE6 && not (isEncoded res) && (bigEnough res)
                 then
-                    case (res, gzipFiles set) of
-                        (ResponseFile s hs file Nothing, GzipCacheFolder cache) ->
-                            case lookup hContentType hs of
-                                Just m
-                                    | gzipCheckMime set m -> compressFile s hs file cache sendResponse
-                                _ -> sendResponse res
-                        _ -> compressE set res sendResponse
+                    let runAction x = case x of
+                            (_ , GzipPreCompressed (GzipPreCompressed _)) -> sendResponse res
```

Comment:
`GzipPreCompressed (GzipPreCompressed ..)` I tried to be "clever" and cut a possible loop. But it is not needed.  
Maybe it would be a bit more transparent to have an explicit list of things we try in order, they will be singletons for the current behavior? Right now I just have a way to try one extra check, without touching any other type.